src/Features/LanguageServer/Protocol/Handler/Definitions/GoToDefinitionHandler.cs

```diff
@@ -18,7 +18,7 @@ public GoToDefinitionHandler(IMetadataAsSourceFileService metadataAsSourceFileSe
         }
 
         public async Task<object> HandleRequestAsync(Solution solution, LSP.TextDocumentPositionParams request,
-            LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken, bool keepThreadContext = false)
+            LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken, bool keepThreadContext = true)
         {
             return await GetDefinitionAsync(solution, request, typeOnly: false, keepThreadContext, cancellationToken).ConfigureAwait(keepThreadContext);
```

Comment:
why use `async/await` here at all?  why not just return `GetDefinitionAsync(solution, request, typeOnly: false, keepThreadContext, cancellationToken)` directly?  Then the caller can decide how this will be called (i.e. with .ConfigureAwait(false/true))